File: src/security-services/src/main/java/com/itechart/security/service/GroupService.java
package com.itechart.security.service;

import com.itechart.security.model.dto.SecuredGroupDto;
import com.itechart.security.model.dto.PublicGroupDto;

import java.io.Serializable;
import java.util.List;

/**
 * @author yauheni.putsykovich
 */
public interface GroupService {

    List<SecuredGroupDto> getGroups();

    Serializable create(SecuredGroupDto group);

    void update(SecuredGroupDto convert);

    SecuredGroupDto get(Long id);

    SecuredGroupDto getGroupWithUsers(Long id);

    List<PublicGroupDto> getPublicGroups();

    void deleteById(Long id);
}

File: src/security-web/src/main/webapp/app/group/group.service.js
/**
 * @author yauheni.putsykovich
 */
angular.module("app").factory("GroupService", ["$http", function ($http) {
    "use strict";
    this.getAll = function () {
        return $http.get("/rest/groups");
    };

    this.getPublicGroups = function () {
        return $http.get("/rest/groups/public");
    };

    this.create = function (group) {
        return $http.post("/rest/groups", group);
    };

    this.update = function (group) {
        return $http.put("/rest/groups", group);
    };

    this.remove = function (id) {
        return $http.delete("/rest/groups/" + id);
    };

    return this;
}]);
